account-kit/smart-contracts/src/ma-v2/client/client.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  createSmartAccountClient,
+  type SmartAccountClient,
+  type SmartAccountSigner,
+  type SmartAccountClientConfig,
+} from "@aa-sdk/core";
+import { type Chain, type CustomTransport, type Transport } from "viem";
+
+import {
+  createSMAV2Account,
+  type CreateSMAV2AccountParams,
+  type SMAV2Account,
+} from "../account/semiModularAccountV2.js";
+
+export type CreateSMAV2AccountClientParams<
+  TTransport extends Transport = Transport,
+  TChain extends Chain = Chain,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = CreateSMAV2AccountParams<TTransport, TSigner> &
+  Omit<
+    SmartAccountClientConfig<TTransport, TChain>,
+    "transport" | "account" | "chain"
+  >;
+
+export function createSMAV2AccountClient<
+  TChain extends Chain = Chain,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>(
+  args: CreateSMAV2AccountClientParams<Transport, TChain, TSigner>
+): Promise<SmartAccountClient<CustomTransport, Chain, SMAV2Account<TSigner>>>;
+
+/**
+ * Creates a MAv2 account client using the provided configuration parameters.
+ *
+ * @example
+ * ```ts
+ * import { http } from "viem";
+ * import { createSMAV2AccountClient } from "@account-kit/smart-contracts";
+ * import { LocalAccountSigner } from "@aa-sdk/core";
+ * import { sepolia } from "@account-kit/infra";
+ *
+ * const MNEMONIC = "...";
+ * const RPC_URL = "...";
+ *
+ * const signer = LocalAccountSigner.mnemonicToAccountSigner(MNEMONIC);
+ *
+ * const chain = sepolia;
+ *
+ * const transport = http(RPC_URL);
+ *
+ * const SMAV2SignerAccountClient = await createSMAV2AccountClient({
+ *  chain,
+ *  signer,
+ *  transport,
+ * });
+ * ```
+ *
+ * @param {CreateSMAV2AccountClientParams} config The configuration parameters required to create the MAv2 account client
+ * @returns {Promise<SmartAccountClient>} A promise that resolves to a `SmartAccountClient` instance
+ */
+export async function createSMAV2AccountClient({
+  ...config
+}: CreateSMAV2AccountClientParams): Promise<SmartAccountClient> {
```

Comment:
No need for that `{ ...config }`, this can just be
```ts
export async function createSMAV2AccountClient(
  config: CreateSMAV2AccountClientParams,
): Promise<SmartAccountClient> {
  // ...
```